docs/source/stateful testing.rst

```diff
@@ -0,0 +1,160 @@
+Stateful Testing
+========================
+
+With Kea’s stateful testing, we can use ``Bundle`` to store some data
+when we want to search for some data manipulation error. In other words,
+Kea tries to generate not just data but entire tests. When we define a set of basic data manipulation actions
+that can be combined, and then Kea will attempt to discover sequences of those actions that lead to a failure.

```

Comment:
Stateful testing is a software testing technique used to validate systems that maintain states over time.
In mobile apps, some functionalities can move from one state to another state based on certain inputs or actions.
Thus, we need additional data structure to support this.

|Here we reference `Hypothesis's Stateful Testing <https://hypothesis.readthedocs.io/en/latest/stateful.html>`_.

In Kea, we can use stateful testing when we write some properties that require stateful information.